tests/build-test-snapd-snap

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash -e
+
+shopt -s nullglob
+
+# check dependencies
+if ! snap list snapcraft >/dev/null; then
```

Comment:
Would we check for channel >= 8.x perhaps even advise stable?

Reply:
I would keep snap out of this.

```sh
SNAPCRAFT=$(command -v snapcraft)
SNAPCRAFT_VERSION=$("$SNAPCRAFT" --version | cut -d ' ' -f 2)

Reply:
Decided against version check, unlikely to be required.